slash/parallel/worker.py

```diff
@@ -3,6 +3,7 @@
 import os
 import pickle
```

Comment:
Is there any test we can add to demonstrate the previous drawbacks we discussed for the old protocol? I'd be more comfortable if there were such a test included...